ibm/service/atracker/resource_ibm_atracker_target.go

```diff
@@ -127,10 +127,15 @@ func ResourceIBMAtrackerTarget() *schema.Resource {
 						},
 						"api_key": &schema.Schema{ // pragma: allowlist secret
 							Type:             schema.TypeString,
-							Required:         true,
+							Optional:         true,
 							Sensitive:        true,
 							DiffSuppressFunc: flex.ApplyOnce,
-							Description:      "The user password (api key) for the message hub topic in the Event Streams instance.",
+							Description:      "The user password (api key) for the message hub topic in the Event Streams instance. This is required if service_to_service is not enabled.",
+						},
+						"service_to_service_enabled": &schema.Schema{
```

Comment:
what is update behavior for this attribute changing the value after initial creation

Reply:
user can switch between plain text password and service to service authentication.  update to service_to_service_enabled cloud turn service to service authentication on or off.